php/media/class-filter.php

```diff
@@ -371,6 +376,52 @@ public function filter_out_local( $content ) {
 		return $this->filter_video_shortcodes( $content );
 	}
 
+	/**
+	 * Apply loading=lazy to <img> if necessary.
+	 *
+	 * @param string $new_tag
+	 * @param bool   $is_admin
+	 *
+	 * @return string
+	 */
+	protected function add_lazy_loading_attribute( $new_tag, $is_admin = false ) {
+		if (
+			apply_filters( 'wp_lazy_loading_enabled', true ) &&
+			false === strpos( $new_tag, 'loading="lazy"' ) &&
+			$is_admin
+		) {
+			$new_tag = str_replace( '/>', ' loading="lazy" />', $new_tag );
+		}
+
+		return $new_tag;
+	}
+
+	/**
+	 * Checks whether $cloudinary_url has f_mp4 which converts gifs to mp4s.
+	 *
+	 * @param string $cloudinary_url
+	 * @param int    $attachment_id
+	 *
+	 * @return string|null
+	 */
+	protected function check_fmp4_presence( $cloudinary_url, $attachment_id ) {
+		$return = null;
+
+		if ( Utils::ends_with( $cloudinary_url, '.gif' ) ) {
+			$all_transformations = Utils::flatten( $this->media->get_transformations_from_string( $cloudinary_url ) );
+
+			if ( in_array( 'mp4', $all_transformations ) ) {
+				$return = sprintf(
+					'<video class="cld-fluid wp-video-%s" autoplay controls src="%s">',
+					$attachment_id,
+					$cloudinary_url
+				);
+			}
```

Comment:
@dugajean @pereirinha  We have methods for finding the index of a transformation:
```suggestion
			$all_transformations = $this->media->get_transformations_from_string( $cloudinary_url );
			$index = $this->media->get_transformation( $all_transformations, 'fetch_format' );
			if( false !== $index && 'mp4' === $all_transformations[ $index ][ 'fetch_format' ] ) {
				$return = sprintf(
					'<video class="cld-fluid wp-video-%s" autoplay controls src="%s">',
					$attachment_id,
					$cloudinary_url
				);
			}
```

However, if we do go with having this as a setting, we can change the check for the presence of the `f_mp4` to the check of the setting.